compressed a lot of code + automatic path (conclusive fix)

DemonixCraft-1 opened the pull request:
Finally compressed to one commit! Officially ready to be merged! Here's what I changed : 
* My accidental install.sh code input from the last fix
* My absurdly large commit history. I compressed my fork down to one 1 commit! If you were to use PR #19 there would have been over 1000 objects to download.
* My really unnecessary code compression for the echo commands in s/sudo
* MINOR FIX : Changed the order of some commands in install.sh
* HOTFIX : Revised ~/.bash_profile procedure and set url to original repo

DemonixCraft committed:
compressed a lot of code + automatic path

purplesyringa commented on install.sh:
```diff
@@ -1,20 +1,6 @@
-echo "enjoy :)"
+echo "Downloading win-sudo...";trap 'echo "Failed to install, sorry :(";exit 1' ERR
+mkdir -p ~/bin/win-sudo && cd "$_";git init -q && git config core.sparsecheckout true;echo s/ >> .git/info/sparse-checkout
+git remote add -mf origin https://github.com/DemonixCraft/win-sudo.git && git pull -q origin master
```
Can you please use `imachug` instead?

DemonixCraft-1 replied:
Yeah I'll put some changes in a new commit

purplesyringa commented on install.sh:
```diff
@@ -1,20 +1,6 @@
+echo "Downloading win-sudo...";trap 'echo "Failed to install, sorry :(";exit 1' ERR
+mkdir -p ~/bin/win-sudo && cd "$_";git init -q && git config core.sparsecheckout true;echo s/ >> .git/info/sparse-checkout
+git remote add -mf origin https://github.com/DemonixCraft/win-sudo.git && git pull -q origin master
+echo "source ~/bin/win-sudo/s/path.sh" | tee -a ~/.bashrc >/dev/null 2>&1 && echo "Win-sudo successfully installed!"
```
Would using `echo "source ~/bin/win-sudo/s/path.sh" >>~/.bashrc && echo "Win-sudo successfully installed!"` instead make any difference or is there a reason we should use `tee`?

DemonixCraft-1 replied:
It would be less safe because it wouldn't ensure that it would be appended to the file but ill check

purplesyringa commented on install.sh:
```diff
@@ -1,20 +1,6 @@
+mkdir -p ~/bin/win-sudo && cd "$_";git init -q && git config core.sparsecheckout true;echo s/ >> .git/info/sparse-checkout
+git remote add -mf origin https://github.com/DemonixCraft/win-sudo.git && git pull -q origin master
+echo "source ~/bin/win-sudo/s/path.sh" | tee -a ~/.bashrc >/dev/null 2>&1 && echo "Win-sudo successfully installed!"
+rm -rf ~/.bash_profile >/dev/null 2>&1;exec bash
```
Not sure about this thing. Probably it would be better if you added `source ~/bin/win-sudo/s/path.sh` line to `bash_profile` as well, because some people may have some scripts here instead of `bashrc`.

purplesyringa replied:
You can include `path.sh` twice - it's okay.

DemonixCraft-1 replied:
tee appends the line to ~/.bashrc

DemonixCraft-1 replied:
I'll look at the options available and see if its possible.

DemonixCraft-1 replied:
~/.bash_profile is usually found in the $root_dir of linux distros. It's generally sourced when your logging in with SSH. It's much more reliable to put the PATH variable in ~/.bashrc because it is sourced upon startup. Putting it in ~/.bash_profile as well means you can activate sudo with one of two ways per shell spawn : 
* Login with SSH
* Run a sourcing command for ~/.bashrc or ~/.bash_profile

My goal here is to make win-sudo completely auto-deployable and only require typing one line of code for complete and quick setup.
 I could either prompt the user to confirm the removal of ~/.bash_profile or temporarily move ~/.bash_profile to a sub dir where it won't conflict with sudo.

DemonixCraft committed:
HOTFIX : Tweaked install.sh

DemonixCraft-1 commented:
Okay here's what I changed in install.sh : 
* Changed url to match your repo
* Set echo command to append to ~/.bashrc, ~/.bash_profile, and /etc/profile, each if existent.
I could not change the echo command because tee is used to append to multiple files as we need here.

purplesyringa reviewed (approved):
Looks like it won't break anything anymore.

imachug commented:
I see, tee makes sense now.

DemonixCraft-1 commented:
Do you want anything else changed?

DemonixCraft-1 commented:
I also have another idea : we could make an alias for `winpty` so we don't have to write it out. 

I was thinking something like ... `alias w='winpty'`
That way we would only have to type `sudo w [command_to_run]`, instead of `sudo winpty [command_to_run]`

imachug commented:
Thank you.
